hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/SyncFuture.java

```diff
@@ -65,102 +74,126 @@
    */
   private Throwable throwable;
 
-  private Thread t;
+
+  /*
+   * Fields below are created once at reset() and accessed without any lock. Should be ok as they
+   * are immutable for this instance of sync future until it is reset.
+   */
+
+  /**
+   * The transaction id of this operation, monotonically increases.
+   */
+  private long txId;
 
   private boolean forceSync;
 
   /**
    * Call this method to clear old usage and get it ready for new deploy.
+   *
    * @param txid the new transaction id
    * @return this
    */
-  synchronized SyncFuture reset(long txid) {
+  SyncFuture reset(long txid, boolean forceSync) {
     if (t != null && t != Thread.currentThread()) {
       throw new IllegalStateException();
     }
     t = Thread.currentThread();
     if (!isDone()) {
       throw new IllegalStateException("" + txid + " " + Thread.currentThread());
     }
+    this.doneLock = new ReentrantLock();
+    this.doneCondition = doneLock.newCondition();
     this.doneTxid = NOT_DONE;
-    this.txid = txid;
+    this.forceSync = forceSync;
+    this.txId = txid;
     this.throwable = null;
     return this;
   }
 
   @Override
-  public synchronized String toString() {
-    return "done=" + isDone() + ", txid=" + this.txid;
+  public String toString() {
+    return "done=" + isDone() + ", txId=" + this.txId;
   }
 
-  synchronized long getTxid() {
-    return this.txid;
+  long getTxId() {
+    return this.txId;
   }
 
-  synchronized boolean isForceSync() {
+  boolean isForceSync() {
     return forceSync;
```

Comment:
This should be covered by the lock? The comment at head of the class says this data member should be covered by the lock? ("all below")